securedrop_client/gui/misc.py

```diff
@@ -0,0 +1,207 @@
+"""
+Generic custom widgets.
+
+Copyright (C) 2018  The Freedom of the Press Foundation.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from typing import Union
+
+from PyQt5.QtCore import QSize, Qt
+from PyQt5.QtWidgets import QHBoxLayout, QLabel, QPushButton, QWidget
+
+from securedrop_client.resources import load_icon, load_svg
+
+
+class SvgToggleButton(QPushButton):
+    """
+    A toggle button used to display the contents of Scalable Vector Graphics (SVG) files provided
+    for an on and off state.
+
+    Parameters
+    ----------
+    on: str
+        The name of the SVG file to add to the button for on state.
+    off: str
+        The name of the SVG file to add to the button for off state.
+    svg_size: QSize, optional
+        The display size of the SVG, defaults to filling the entire size of the widget.
+    """
+
+    def __init__(self, on: str, off: str, svg_size: str = None):
+        super().__init__()
+        print("OK")
+
+        # Set layout
+        layout = QHBoxLayout(self)
+        self.setLayout(layout)
+
+        # Remove margins and spacing
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+
+        # Add SVG icon and set its size
+        self._icon = load_icon(normal=on, normal_off=off)
```

Comment:
note: this class was copied exactly except for adding the `_` here and with other private members.

Reply:
I removed this from the commit. It really did belong in the commit after when using `button.icon()` (part of the `QPushButton` API) in the test file.